infra/main.bicep

```diff
@@ -82,11 +87,15 @@ module api './app/api.bicep' = {
     name: !empty(apiContainerAppName) ? apiContainerAppName : '${abbrs.appContainerApps}${apiServiceName}-${resourceToken}'
     location: location
     imageName: apiImageName
-    applicationInsightsName: monitoring.outputs.applicationInsightsName
     containerAppsEnvironmentName: appEnv.outputs.environmentName
     containerRegistryName: appEnv.outputs.registryName
     serviceName: apiServiceName
+    managedIdentityName: security.outputs.managedIdentityName
   }
+  dependsOn: [
```

Comment:
Same goes for this example. We should try to avoid using dependsOn when possible. Using outputs in a module is another way of setting up dependency chaining.